Lib9c/Action/Reregister.cs

```diff
@@ -0,0 +1,197 @@
+            {
+                throw new FailedLoadStateException($"{addressesHex} failed to load {nameof(OrderDigest)}({orderReceiptAddress}).");
+            }
+            var digestList = new OrderDigestListState(rawList);
+            digestList.Remove(orderOnSale.OrderId);
+            states = states.SetState(itemAddress, itemOnSale.Serialize())
+                .SetState(shopAddress, shardedShopState.Serialize())
+                .SetState(orderReceiptAddress, digestList.Serialize());
+            sw.Stop();
+            
+            var expirationMail = avatarState.mailBox.OfType<OrderExpirationMail>()
+                            .FirstOrDefault(m => m.OrderId.Equals(orderId));
+            if (!(expirationMail is null))
+            {
+                avatarState.mailBox.Remove(expirationMail);
+            }
+
+            // for reregister
+            var reregisterShopState = states.TryGetState(reregisterShopAddress, out Dictionary serializedState)
+                ? new ShardedShopStateV2(serializedState)
+                : new ShardedShopStateV2(reregisterShopAddress);
+
+            Log.Verbose("{AddressesHex} Reregister Get ShardedShopState: {Elapsed}", addressesHex, sw.Elapsed);
+            sw.Restart();
+            var newOrder = OrderFactory.Create(context.Signer, sellerAvatarAddress, reregisterOrderId, price, tradableId,
+                context.BlockIndex, itemSubType, count);
+            newOrder.Validate(avatarState, count);
+
+            var tradableItem = newOrder.Sell(avatarState);
+            var costumeStatSheet = states.GetSheet<CostumeStatSheet>();
+            var orderDigest = newOrder.Digest(avatarState, costumeStatSheet);
+            reregisterShopState.Add(orderDigest, context.BlockIndex);
+
+            digestList.Add(orderDigest);
+            states = states.SetState(orderReceiptAddress, digestList.Serialize());
+            states = states.SetState(inventoryAddress, avatarState.inventory.Serialize())
+                .SetState(worldInformationAddress, avatarState.worldInformation.Serialize())
+                .SetState(questListAddress, avatarState.questList.Serialize())
+                .SetState(sellerAvatarAddress, avatarState.SerializeV2());
+            sw.Stop();
+
+            Log.Verbose("{AddressesHex} Reregister Set AvatarState: {Elapsed}", addressesHex, sw.Elapsed);
+            sw.Restart();
+            states = states
+                .SetState(itemAddress, tradableItem.Serialize())
+                .SetState(reregisterOrderAddress, newOrder.Serialize())
+                .SetState(reregisterShopAddress, reregisterShopState.Serialize());
+            sw.Stop();
+
+            var ended = DateTimeOffset.UtcNow;
+            Log.Verbose("{AddressesHex} Reregister Set ShopState: {Elapsed}", addressesHex, sw.Elapsed);
+            Log.Verbose("{AddressesHex} Reregister Total Executed Time: {Elapsed}", addressesHex, ended - started);
```

Comment:
How about use another `Stopwatch` to calculate whole time.

Reply:
Caching DateTimeOffset.UtcNow in `started` `ended`. `ended - started' seems to be the total time